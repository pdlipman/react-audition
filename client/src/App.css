.App {
  text-align: center;
  color: rgb(33, 150, 243);
}

.App-header {
  background-color: rgb(33, 150, 243);
  padding: 20px;
  color: white;
}

.App-title {
  font-size: 1.5em;
}

.App-intro {
  font-size: large;
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}

.LeftItem {
  text-align: left;
}

.RightItem {
  text-align: right;
}

.DownArrow {
  margin-top: 5px;
  display: inline-block;
  vertical-align: middle;
}

.DownArrowOverlay {
  margin-left: -24px;
}

.UpArrow {
  margin-bottom: 5px;
  display: inline-block;
  vertical-align: middle;
}

.Expansion {
  border: 1px solid rgb(33, 150, 243);
}
