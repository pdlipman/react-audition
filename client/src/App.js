import React, { Component } from 'react';

import {
  createStore,
  applyMiddleware,
  compose,
} from 'redux';
import { Provider } from 'react-redux';
import thunkMiddleware from 'redux-thunk';

import combinedReducer from './combinedReducers';

import Dashboard from './components/dashboard/Dashboard';
import './App.css';

const store = createStore(
  combinedReducer,
  compose(
    applyMiddleware(thunkMiddleware),
    window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__(),
  ),
);

class App extends Component {
  render() {
    return (
      <Provider store={ store }>
        <div className="App">
          <header className="App-header">
            <h1 className="App-title">Front Porch Interview Project</h1>
          </header>
          <p className="App-intro">
            Some boilerplate code has been provided; please build upon it (or replace it) to create your solution.
          </p>
          <p className="App-intro">Good luck!</p>
          <Dashboard/>
        </div>
      </Provider>
    );
  }
}

export default App;
