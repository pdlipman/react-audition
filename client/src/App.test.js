import React from 'react';
import Immutable from 'immutable';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import { shallow } from 'enzyme';
import App from './App';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);
const testState = Immutable.fromJS({});

describe('App', () => {
  test('App should render properly', () => {
    const store = mockStore(testState);
    const wrapper = shallow(
      <App store={ store }/>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
