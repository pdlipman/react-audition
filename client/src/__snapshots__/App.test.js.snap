// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App should render properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    store={
        Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div
        className="App"
>
        <header
                className="App-header"
        >
                <h1
                        className="App-title"
                >
                        Front Porch Interview Project
                </h1>
        </header>
        <p
                className="App-intro"
        >
                Some boilerplate code has been provided; please build upon it (or replace it) to create your solution.
        </p>
        <p
                className="App-intro"
        >
                Good luck!
        </p>
        <Connect(Dashboard) />
</div>,
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className="App-header"
>
            <h1
                        className="App-title"
            >
                        Front Porch Interview Project
            </h1>
</header>,
          <p
            className="App-intro"
>
            Some boilerplate code has been provided; please build upon it (or replace it) to create your solution.
</p>,
          <p
            className="App-intro"
>
            Good luck!
</p>,
          <Connect(Dashboard) />,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1
              className="App-title"
>
              Front Porch Interview Project
</h1>,
            "className": "App-header",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Front Porch Interview Project",
              "className": "App-title",
            },
            "ref": null,
            "rendered": "Front Porch Interview Project",
            "type": "h1",
          },
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Some boilerplate code has been provided; please build upon it (or replace it) to create your solution.",
            "className": "App-intro",
          },
          "ref": null,
          "rendered": "Some boilerplate code has been provided; please build upon it (or replace it) to create your solution.",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Good luck!",
            "className": "App-intro",
          },
          "ref": null,
          "rendered": "Good luck!",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className="App"
>
          <header
                    className="App-header"
          >
                    <h1
                              className="App-title"
                    >
                              Front Porch Interview Project
                    </h1>
          </header>
          <p
                    className="App-intro"
          >
                    Some boilerplate code has been provided; please build upon it (or replace it) to create your solution.
          </p>
          <p
                    className="App-intro"
          >
                    Good luck!
          </p>
          <Connect(Dashboard) />
</div>,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <header
              className="App-header"
>
              <h1
                            className="App-title"
              >
                            Front Porch Interview Project
              </h1>
</header>,
            <p
              className="App-intro"
>
              Some boilerplate code has been provided; please build upon it (or replace it) to create your solution.
</p>,
            <p
              className="App-intro"
>
              Good luck!
</p>,
            <Connect(Dashboard) />,
          ],
          "className": "App",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <h1
                className="App-title"
>
                Front Porch Interview Project
</h1>,
              "className": "App-header",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Front Porch Interview Project",
                "className": "App-title",
              },
              "ref": null,
              "rendered": "Front Porch Interview Project",
              "type": "h1",
            },
            "type": "header",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Some boilerplate code has been provided; please build upon it (or replace it) to create your solution.",
              "className": "App-intro",
            },
            "ref": null,
            "rendered": "Some boilerplate code has been provided; please build upon it (or replace it) to create your solution.",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Good luck!",
              "className": "App-intro",
            },
            "ref": null,
            "rendered": "Good luck!",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
